yarn-project/aztec/src/cli/cmds/start_node.ts

```diff
@@ -42,11 +43,18 @@ export async function startNode(
     } else {
       throw new Error('--node.publisherPrivateKey or --l1-mnemonic is required to deploy L1 contracts');
     }
+    // REFACTOR: We should not be calling a method from sandbox on the prod start flow
     await deployContractsToL1(nodeConfig, account!, undefined, {
       assumeProvenThroughBlockNumber: nodeSpecificOptions.assumeProvenThroughBlockNumber,
       salt: nodeSpecificOptions.deployAztecContractsSalt,
     });
   }
+  // If not deploying, validate that the addresses and config provided are correct.
+  // Eventually, we should be able to dynamically load this just by having the L1 governance address,
+  // instead of only validating the config the user has entered.
+  else {
+    await validateL1Config({ ...getNodeConfigEnvVars(), ...nodeConfig });
```

Comment:
same here, did you see it not working and added the `get{service}EnvVars`?

Reply:
I found some config vars were not being populated without calling this. Eg if I add, right before this call:

```
    userLog(`Governance from node config: ${nodeConfig?.l1Contracts?.governanceAddress} `);
    userLog(`Governance from env vars: ${{ ...getNodeConfigEnvVars(), ...nodeConfig }.l1Contracts?.governanceAddress}`);
```

And then I run:

```
GOVERNANCE_CONTRACT_ADDRESS=0x976EA74026E726554dB657fA54763abd0C3a0aa9 [...] yarn aztec start --node --archiver --sequencer --pxe
```

I get:
```
Governance from node config: undefined 
Governance from env vars: 0x976ea74026e726554db657fa54763abd0c3a0aa9
```

It seems there's a need for cleaning up init config. But I didn't want to go there in this PR.